### PR TITLE
React - Introduction to State #40431

### PR DESCRIPTION
**Pull request by shashankbaranwal**
1. Create a Slider component in a similar way like TextField component.
2. Install “prop-types” package
3. The component must accept the following props:
* altText: Default values are *Default Banner”, and will be used for *img* alt tag
* banners: Array containing the path of images
* defaultBanner: String, default image is”default.png”, will be used in case banners are not provided.
* duration: Number, the Default value is 2000, Rotation of Banner image in milliseconds.
* height: Number, the default is 200, and will set the height of the slider.
* random: Boolean, default is false if true then images will come in random order, otherwise images will render in Round Robin fashion.
Use defaultProps and propsTypes to set up these.
4. Create a filer “lib/utils/math.js”, and inside this file export following functions:
* getRandomNumber - accept a max number and will return the number b/w 0 to number - 10
* getNextRoundRobin - accept two arguments - total and current, and will return the next number in the round robin fashion e.g. say the total is 10, and current is 7, then it will return 8, if the current is 9, then it will return 0.
5. Use the above two functions to calculate the next banner index depending upon the value of random props.
6. Create a file configs/constants.js and then export following constants from them and use it inside Slider component, e.g.

export const PUBLIC_IMAGE_FOLDER = '/images/';
export const DEFAULT_BANNER_IMAGE = 'banners/default.png';

7. Initialize the SetInterval inside componentDidMount, and do deinitialize it inside componentWIllUnmount.
8. Please make sure that all the images will maintain their aspect ratio, and images must be center aligned.
9. Use this component inside “TextFieldDemo” components with various flavors.
10. Please seem Banner.gif for your reference.

Note:
1. Don’t make your render function too long.
2. Please follow DRY principle.
3. Do follow your Airbnb Guidelines.

### CODE REVIEW

File: src/components/Slider/Slider.jsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  componentWillUnmount = () => {
+    clearInterval(this.id);
+  }
+
+  render() {
+    const { current } = this.state;
+    const {
+      altText, height, duration, banner,
+    } = this.props;
+    const { defaultbanner } = this.props;
```
Comment (namanpar) [inline src/components/Slider/Slider.jsx:38]: destructure defaultbanner along with altTest and others. No need to destructure twice from props. Use camelCase for defaultbanner as it is also given ticket

File: src/components/Slider/Slider.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        </div>
+      </>
+    );
+  }
+}
+export default Slider;
+Slider.propTypes = {
+  altText: PropTypes.string,
+  banner: PropTypes.arrayOf(PropTypes.string),
+  defaultbanner: PropTypes.string,
```
Comment (namanpar) [inline src/components/Slider/Slider.jsx:61]: use camelCase

**Review by namanpar — CHANGES_REQUESTED** [review]

File: src/libs/utils/math.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+function getRandomNumber(max) {
+  return (Math.floor(Math.random() * Math.floor(max - 1)));
+}
+function getNextRoundRobin(current, total) {
+  return (current === total - 1 ? 0 : current + 1);
+}
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/libs/utils/math.js]: Used ES6

File: src/components/Slider/Slider.jsx (hunk 1 above)
Comment (shashankbaranwal) [reply src/components/Slider/Slider.jsx]: destructured the defaultBanner along with altTest and others. Used camelCase for defaultBanner

File: src/components/Slider/Slider.jsx (hunk 2 above)
Comment (shashankbaranwal) [reply src/components/Slider/Slider.jsx]: Used camelCase

File: src/components/Slider/Slider.jsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      current: -1,
+
+    };
+  }
+
+  componentDidMount = () => {
+    const { random, duration } = this.props;
+    let { current } = this.state;
+    this.id = setInterval(() => {
+      if (random) {
```
Comment (namanpar) [inline src/components/Slider/Slider.jsx:20]: use ternary instead of if-else

**Review by namanpar — CHANGES_REQUESTED** [review]

File: src/components/Slider/Slider.jsx (hunk 4 above)
Comment (shashankbaranwal) [reply src/components/Slider/Slider.jsx]: Used ternary instead of if-else

**Review by namanpar — APPROVED** [review]

**Pull request merged**